### ISSUE TITLE
Differences between 2.1.1 keyboard guidance in WCAG2ICT and EN 301 549 content in clause 11.2.1.1 Keyboard

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#691]
The EN 301 549 made changes to the WCAG2ICT 2.1.1 Keyboard note 2 for in clause 11.2.1.1. Want to have discussion regarding their changed text to ensure we agree and whether WCAG2ICT should incorporate any changes due to this.

WCAG2ICT's Note 2:
>NOTE 2 (Added): This success criterion does not imply that software always needs to directly support a keyboard or “keyboard interface”. Nor does it imply that software always needs to provide a [virtual keyboard](https://w3c.github.io/wcag2ict/#virtual-keyboard).

Link to the WCAG2ICT full guidance: [Applying 2.1.1 Keyboard to non-web documents and software](https://w3c.github.io/wcag2ict/#applying-sc-2-1-1-keyboard-to-non-web-documents-and-software).

EN 301 549 split into 2 notes and added text:
>NOTE 2: This success criterion does not imply that software always needs to directly support a keyboard or "keyboard interface" if one is not provided by the platform software. But if one is provided, the software needs to make all functionality available through it - unless the exception applies.
>
>NOTE 3: Nor does it imply that software always needs to provide its own virtual keyboard. But if it does, then it also needs to also support keyboard input from any keyboard interface provided by platform software.

**Comment by GreggVan** [comment]
We need to change WCAG2ICT notes to what is in EN 301 549.   I didnt  notice before but the WCAG2ICT note contradicts the SC and its intent.  EN 301 549 language much better

**Comment by GreggVan** [comment]
if you look at  11.2.1.1  you will see that 

a)  it only applies to ATMs that have a keyboard or keyboard interface  - so it would be no problem
and 
b) if an ATM does not provide a keyboard or keyboard interface then it automatically satisfies 11.2.1.1 

> Where ICT is, or includes, [non-web software] that provides a [user interface], and that provides a keyboard or accepts input from a keyboard or keyboard interface, 
> the software shall satisfy the [WCAG 2.2 Success Criterion 2.1.1 Keyboard].

**NOW  - what do we put in WCAG2ICT???**

I  suggest the following  (**_since our current proposed note cannot be used since it contradicts the provision_** - limiting it - which we cannot do in a note.

FIX WCAG2ICT - not using a note - but by changing our advice to 

> _Applying SC 2.1.1 Keyboard to Non-Web Documents and Software_
> 
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent) **if it is limited to ICT that is, or includes, non-web software that provides a keyboard or accepts input from a keyboard or keyboard interface** .
>

**Comment by GreggVan** [comment]
Note 1 (Added): Keyboard interface does not refer to a physical device but to the platform service that  provides software with keystrokes from all keyboards or keyboard substitutes. When software supports such a device-independent service of the platform, and the software or non-web document functionality is made fully operable through the service, then this success criterion would be satisfied.

**Comment by loicmn** [comment]
+1 to Gregg's proposal

**Comment by mapluke** [comment]
Also +1 to Gregg's proposal.

**Comment by maryjom** [comment]
Not sure we can add to the "This applies directly as written..." suggested changes:
>This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent) if it is limited to ICT that is, or includes, non-web software that provides a keyboard or accepts input from a keyboard or keyboard interface.

Software doesn't provide a keyboard - it's ICT itself (the hardware). IMO, closed functionality is a weird case. They might have a numeric keypad and function keys only - is that considered a keyboard?

**Comment by maryjom** [comment]
Wondering if we could introduce a non-web software precondition such as: 

>Where the non-web software runs on ICT that includes a keyboard with a software platform that provides device-independent input services for keyboard operation, this applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent).

**Comment by GreggVan** [comment]
@maryjom   All that you write is correct.  And it all occurs through the "keyboard interface".   Key events provide the key-codes for the key presses - via the "keyboard interface".   So software that is operable using only input from the keyboard interface..... 

so no change.

**Comment by GreggVan** [comment]
@maryjom 

We can re-introduce a precondition but the one you post is not complete -   Here is the precondition.

>Where ICT is or includes non-web software that can be run on a software platform that provides a **_keyboard interface_** service,  this applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent).

**Comment by maryjom** [comment]
Suggest a slight modification to use consistent terms:

>Where ICT is or includes non-web software that can be run on a software platform that provides a **device-independent** keyboard interface service, this applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent).

**Comment by maryjom** [comment]
In my haste, I neglected to have the TF discuss the potential to add a pre-condition with text being proposed.  Since the TF did reach resolution on the changes to the notes (in the 3 July and 17 July meetings), I'm inclined to open a new issue and PR regarding the pre-condition so I can incorporate the agreed notes this week and, for now, leave the existing text in there that says:
>###### Applying SC 2.1.1 Keyboard to Non-Web Documents and Software
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard#intent).
